pkg/istio-agent/agent_test.go

```diff
@@ -489,6 +489,39 @@ func TestAgent(t *testing.T) {
 
 		testutil.CompareContent(t, got, filepath.Join(env.IstioSrc, "pkg/istio-agent/testdata/grpc-bootstrap.json"))
 	})
+	t.Run("ROOT CA change", func(t *testing.T) {
+		dir := mktemp()
+		rootCertFileName := "root-cert.pem"
+
+		// use a invalid root cert, XDS will fail with `authentication handshake failed`
+		localRootCert := filepath.Join(env.IstioSrc, "./tests/testdata/local/etc/certs/root-cert.pem")
+		if err := file.Copy(localRootCert, dir, rootCertFileName); err != nil {
+			t.Fatalf("failed to init root CA: %v", err)
+		}
+		a := Setup(t, func(a AgentTest) AgentTest {
+			a.AgentConfig.XDSRootCerts = path.Join(dir, rootCertFileName)
+			return a
+		})
+		meta := proxyConfigToMetadata(t, a.ProxyConfig)
+		err := retry.Until(func() bool {
+			err := test.Wrap(func(t test.Failer) {
+				conn := setupDownstreamConnectionUDS(t, a.AgentConfig.XdsUdsPath)
+				xdsc := xds.NewAdsTest(t, conn).WithMetadata(meta)
+				_ = xdsc.RequestResponseAck(t, nil)
+			})
+
+			return err == nil
+		}, retry.Timeout(time.Second*5), retry.Delay(time.Millisecond*200))
```

Comment:
This is not great since it means the test always takes at LEAST 5s. IIUC, we are expecting a failure, so do we really need to check it failed 5s/200ms=25x in a row? Seems like we can just bring the test.Wrap outside of retry.Until? Then its one attempt and very quick

Reply:
agree that.